internal/table/console.go

```diff
@@ -0,0 +1,104 @@
+package table
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/mattn/go-runewidth"
+	console "github.com/nathan-fiscaletti/consolesize-go"
+)
+
+var consoleSize = console.GetConsoleSize
+
+const (
+	alignLeft = iota
+	alignCenter
+)
+
+type TabularWriter struct {
+	out         io.Writer
+	rowsWritten int
+	rowCount    int
+	columnCount int
+}
+
+func NewTableWriter(out io.Writer) *TabularWriter {
+	return &TabularWriter{out: out}
+}
+
+func (t *TabularWriter) Write(cells ...interface{}) error {
+	t.initSize(cells)
+	// colWidth = (totalWidth - numOfSeparators) / numOfColumns
+	colWidth := (t.columnCount - (len(cells)+1)*2) / len(cells)
+	line := makeTabularLine(colWidth, alignLeft, cells...)
+	if _, err := fmt.Fprintln(t.out, line); err != nil {
+		return err
+	}
+	t.rowsWritten++
+	if t.rowsWritten == t.rowCount {
+		// reset state for a new page
+		t.rowsWritten = 0
+	}
+	return nil
+}
+
+func (t *TabularWriter) initSize(cells []interface{}) {
+	if t.rowsWritten != 0 {
+		// not start state
+		return
+	}
+	// start state
+	t.columnCount, t.rowCount = consoleSize()
+	// minimum space required for => | ... | ... | ... |
+	if t.columnCount < (len(cells)+1)*2+len(cells)*3 {
+		t.columnCount = (len(cells)+1)*2 + len(cells)*3
+	}
+	if t.rowCount < 4 {
+		t.rowCount = 4
+	}
+}
+
+/*
+WriteHeader outputs header for the table with the form:
++--------------------------------------+
+| vegetables | fruit      | rank       |
++--------------------------------------+
+*/
+func (t *TabularWriter) WriteHeader(cells ...interface{}) error {
```

Comment:
Just wondering, will you plan to use same function to produce output files or, what if I direct `>` the output to a file? In this case, how trimmed columns (`longlongvegetables` -> `longlongvege...`) will be managed.

Reply:
Great catch! I imagine it is not a use case for interactive browsing. This would be very useful for non-interactive scripting purposes. For that we are considering adding a "--csv" flag such as:
```hzc sql query --csv "SQLQUERY" ```
which will output every field and column without applying any wrapping.

And for now, I do not think we should support it on interactive mode since we are trying to separate those use cases.

Reply:
I got it, thanx. Than, there is no problem. While talking about flags, can we able to set flags also from environment variable?

Reply:
Currently we do not have support for that. I think it is low effort to do, but I am not sure how common it is on linux environment to manipulate flags via environment variable.
If you come across something like that, I think it would be great to consider it 🙏